vindex/map.go

```diff
@@ -62,11 +63,23 @@ type MapFn func([]byte) [][sha256.Size]byte
 type InputLog interface {
 	// Checkpoint returns the latest checkpoint committing to the input log state.
 	Checkpoint(ctx context.Context) (checkpoint []byte, err error)
+	// Parse unmarshals and verifies a checkpoint obtained from GetCheckpoint.
+	Parse(checkpoint []byte) (*log.Checkpoint, error)
```

Comment:
Could you get away with just having `Checkpoint()` which did the validation and parsing too (maybe returning `[]byte` and `*log.Checkpoint` if you need both in the caller)?